server.ts

```diff
@@ -0,0 +1,57 @@
+import * as express from 'express';
+import { TodoApp } from './src/todo';
+
+const todo = new TodoApp();
+
+const serv = express();
+serv.use(express.json());
+
+const port = 3000;
+
+serv.get('/', function (req, res) {
+    res.send(todo.listAllToDos());
+});
+
+serv.put('/', function (req, res) {
+    const data = req.body;
+    try {
+        todo.addItem(data.description, data.dueAt);
+        res.sendStatus(200);
+    } catch (e) {
+        if (e.message = 'The date you entered is invalid') {
```

Comment:
A single `=` operator is an assignment (definitely not what you want here).  `==` is a weak type comparison; `===` is a strong type comparison.  In this case you normally want to be using strong type comparisons.

Reply:
This applies throughout...